source/Calamari.AzureAppService/Behaviors/AzureAppServiceBehaviour.cs

```diff
@@ -115,35 +116,57 @@ public async Task Execute(RunningDeployment context)
              * https://github.com/OctopusDeploy/Calamari/tree/master/source/Calamari.Common/Features/Behaviours
              */
             var uploadPath = string.Empty;
-            if (substitutionFeatures.Any(featureName => context.Variables.IsFeatureEnabled(featureName)))
-                uploadPath = (await Archive.PackageArchive(context.StagingDirectory, context.CurrentDirectory)).FullName;
-            else
-                uploadPath = (await Archive.ConvertToAzureSupportedFile(packageFileInfo)).FullName;
+            bool uploadFileNeedsCleaning = false;
+            try
+            {
+                if (substitutionFeatures.Any(featureName => context.Variables.IsFeatureEnabled(featureName)))
+                {
+                    uploadPath = (await Archive.PackageArchive(context.StagingDirectory, context.CurrentDirectory)).FullName;
+                    uploadFileNeedsCleaning = false;
+                }
+                else
+                {
+                    var uploadFile = await Archive.ConvertToAzureSupportedFile(packageFileInfo); 
+                    uploadPath = uploadFile.FullName;
+                    uploadFileNeedsCleaning = uploadFile.Extension != packageFileInfo.Extension; // did the file get changed
```

Comment:
There are three implementations of the package providers, only the nuget one makes any changes to the file for uploading.

Reply:
There's also the LegazyAzureAppServiceBehaviour that follows this same code path. 
Can the file deletion be pushed into the NugetPackageProvider to handle both?

Reply:
Scrap that this is 2023.3. LGTM